### ISSUE TITLE
Message correlation works only for one tenant if the same correlationId is used with multiple tenants

### ISSUE THREAD

**Issue by mustafadagher** [camunda/camunda#25509]
**Describe the bug**

As reported by a [support case](https://jira.camunda.com/browse/SUPPORT-24335), there seems to be a problem with correlating a message to start a new process when you have multiple tenants that have the same process definitions if you use the same `correlationKey`.

When reproduced by support: the following was found:
> It seems that if you have an active process instance which was correlated by the same key, then - as long as the process instance is running - no other subscriptions can be correlated with the same correlation key. This applies even if the two different subscriptions are on different tenants. Once the process instance is finished the other message can be correlated too.

I looked into the message correlation logic. We intend to create only one subscription per correlation key..
However, I see that in [MessageCorrelateBehavior.correlateToMessageStartEvents](https://github.com/camunda/camunda/blob/59845f6e909dc9f2c02e76ab6842f62181a74180/zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/message/MessageCorrelateBehavior.java#L56) we pass the `tenantId` to the [messageState.existActiveProcessInstance](https://github.com/camunda/camunda/blob/59845f6e909dc9f2c02e76ab6842f62181a74180/zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/message/DbMessageState.java#L362-L370), but the tenant id is never used and it's not part of the column family or the composite key.

I also found this [comment](https://github.com/camunda/camunda/pull/14356#pullrequestreview-1635075706) on a fairly old PR, expecting the tenantId usage to be included in a subsequent [PR](https://github.com/camunda/camunda/pull/14450)
Which did not include marking `activeProcessInstancesByCorrelationKeyColumnFamily` or `processInstanceCorrelationKeyColumnFamily` as `TenantAware`

I can't personally decide if this was an intended behavior or is it a bug resulting forgetting to support tenant awareness for this part.

**Expected behavior**

One tenant should not be able to affect the execution behavior of another tenant. The correlation should work across different tenants even when using the same `correlationkey`

**Environment:**
- Zeebe Version: 8.5.8

**Comment by remcowesterhoud** [comment]
I agree it's an oversight.

The workaround would be to redeploy the process so it's a different id, or to use a different correlation key.

**Comment by korthout** [comment]
Severity mid as there is a workaround: use different process ids or correlation keys. Likelihood low, as this as has been around since multi-tenancy was released with 8.2 and this is the first time we encounter it. I don't expect many customers to encounter this.

If we do notice this occurring more often, we can increase the likelihood. If the workaround cannot be used, please let us know your case so we can adjust the severity.
